### PR TITLE
Rename module resolution option `node` to `node10` (preserving backward-compatible alias)

### PR DESCRIPTION
**Pull request by andrewbranch**
Part of #50152 

This does not make it an error to specify `--moduleResolution node`, but does ensure that if we ever mention the option, we call it `node10`. I’m going to see if we can give another push away from `node` when editing tsconfig files in VS Code as well. (I had initially proposed removing `node` from the JSON schema, but ideally we would want to do that only if the user’s TypeScript version is 5.0+, at least until that’s widely adopted.) That’s probably the extent of the deprecation warnings/errors we’ll be able to give.

### CODE REVIEW

**Comment by andrewbranch** [comment]
Likely after the holidays, I also want to discuss a process for transitioning various defaults away from `node10`.

File: src/compiler/commandLineParser.ts [hunk 1]
```diff
@@ -961,12 +961,15 @@ const commandOptionsWithoutBuild: CommandLineOption[] = [
...
-            node: ModuleResolutionKind.NodeJs,
+            // N.B. The first entry specifies the value shown in `tsc --init`
+            node10: ModuleResolutionKind.Node10,
+            node: ModuleResolutionKind.Node10,
             classic: ModuleResolutionKind.Classic,
             node16: ModuleResolutionKind.Node16,
             nodenext: ModuleResolutionKind.NodeNext,
             bundler: ModuleResolutionKind.Bundler,
         })),
+        deprecatedKeys: new Set(["node"]),
```
Comment (andrewbranch) [inline src/compiler/commandLineParser.ts:972]: Right now, this only filters the config parsing diagnostic that says what the valid options are.

**Review by RyanCavanaugh — APPROVED** [review]

**Comment by andrewbranch** [comment]
Somehow I failed to update a big batch of baselines with this kind of change

```diff
-     "Module resolution kind is not specified, using 'NodeJs'.",
+     "Module resolution kind is not specified, using 'Node10'.",
```

File: src/compiler/diagnosticMessages.json [hunk 2]
```diff
@@ -3391,7 +3391,7 @@
         "category": "Error",
         "code": 2791
     },
-    "Cannot find module '{0}'. Did you mean to set the 'moduleResolution' option to 'node', or to add aliases to the 'paths' option?": {
+    "Cannot find module '{0}'. Did you mean to set the 'moduleResolution' option to 'nodenext', or to add aliases to the 'paths' option?": {
```
Comment (andrewbranch) [inline src/compiler/diagnosticMessages.json:3394]: I also meant to leave some commentary on this change. This message was written when there were only two module resolution options, and only one worked with non-relative specifiers. Now that there’s `node16`, `nodenext`, and `bundler`, it’s a bit harder to infer what someone on `classic` should be on instead. I picked `nodenext`, but we could revisit this message in the future if we need to.

**Pull request merged**